test/integration/framework.go

```diff
@@ -109,24 +120,45 @@ func withConfigGetFreshApiserverAndClient(
 		options.SecureServingOptions.ServerCert.CertDirectory = certDir
 
 		if err := server.RunServer(options, stopCh); err != nil {
-			close(serverFailed)
-			t.Logf("Error in bringing up the server: %v", err)
+			for err != nil {
```

Comment:
I would add some sort of limit here just to prevent tests from hanging in CI somewhere until someone comes to look at them. 10 should be more than enough?

Reply:
Thanks for the review @jberkhahn, I agree.  With the find from @MHBauer we get rid of this kludginess and let the OS pick the port and no more retries.